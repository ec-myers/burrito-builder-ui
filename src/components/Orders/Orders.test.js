import React from 'react';
import { shallow } from 'enzyme';
import * as actions from '../../actions';
// import { getOrders } from '../../apiCalls';
import { Orders, mapStateToProps, mapDispatchToProps } from './Orders';


describe('Orders', () => {
  let wrapper;
  let orders;
  jest.mock('../../actions');
  // jest.mock('../../apiCalls');

  beforeEach(() => {
    orders = [
      {
        id: 1,
        name: 'Emmett',
        ingredients: ['beans', 'cheese']
      },
      {
        id: 2,
        name: 'Penelope',
        ingredients: ['chicken', 'cheese']
      }
    ];
    wrapper = shallow(<Orders orders={orders}/>)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return an object with 1 property', () => {
      let orders;
      let mockState = {
        orders
      };

      let expected = {
        orders
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    let mockDispatch;

    beforeEach(() => {
      mockDispatch = jest.fn();
    });

    it('should call dispatch with a setOrders action', () => {
      let orders = [
        {
          id: 1,
          name: 'Emmett',
          ingredients: ['beans', 'cheese']
        },
        {
          id: 2,
          name: 'Penelope',
          ingredients: ['chicken', 'cheese']
        }
      ];

      const actionToDispatch = actions.setOrders(orders);
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.setOrders(orders);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });

});