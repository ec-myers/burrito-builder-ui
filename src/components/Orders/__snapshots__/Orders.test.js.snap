// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders should match the snapshot 1`] = `
<section>
  <div
    className="order"
    key="1"
  >
    <h3>
      Emmett
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="beans"
      >
        beans
      </li>
      <li
        key="cheese"
      >
        cheese
      </li>
    </ul>
  </div>
  <div
    className="order"
    key="2"
  >
    <h3>
      Penelope
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="chicken"
      >
        chicken
      </li>
      <li
        key="cheese"
      >
        cheese
      </li>
    </ul>
  </div>
</section>
`;
